src/redprl/machine.fun

```diff
@@ -46,28 +46,26 @@ struct
   val todo = Fail "TODO"
   fun ?e = raise e
 
-  local
-    fun plug m = 
-      fn APP (HOLE, n) => Syn.intoApp (m, n)
-       | HCOM (dir, HOLE, cap, tubes) => Syn.intoHcom {dir = dir, ty = m, cap = cap, tubes = tubes}
-       | COE (dir, (u, HOLE), coercee) => Syn.intoCoe {dir = dir, ty = (u, m), coercee = coercee}
-       | IF ((x, tyx), HOLE, t, f) => Syn.into @@ Syn.IF ((x, tyx), m, (t, f))
-       | S1_REC ((x, tyx), HOLE, base, (u, loop)) => Syn.into @@ Syn.S1_REC ((x, tyx), m, (base, (u, loop)))
-       | DIM_APP (HOLE, r) => Syn.into @@ Syn.DIM_APP (m, r)
-       | NAT_REC ((z, tyz), HOLE, zer, (x, y, succ)) => Syn.into @@ Syn.NAT_REC ((z, tyz), m, (zer, (x, y, succ)))
-       | INT_REC ((z, tyz), HOLE, zer, (x,y,succ), negone, (x',y',negss)) => Syn.into @@ Syn.INT_REC ((z, tyz), m, (zer, (x,y,succ), negone, (x',y',negss)))
-       | PROJ (lbl, HOLE) => Syn.into @@ Syn.PROJ (lbl, m)
-       | TUPLE_UPDATE (lbl, n, HOLE) => Syn.into @@ Syn.TUPLE_UPDATE ((lbl, m), m)
-       | PUSHOUT_REC ((x, tyx), HOLE, (y, left), (z, right), (u, w, glue)) => Syn.into @@ Syn.PUSHOUT_REC ((x, tyx), m, ((y, left), (z, right), (u, w, glue)))
-       | COEQUALIZER_REC ((x, tyx), HOLE, (y, cod), (u, w, dom)) => Syn.into @@ Syn.COEQUALIZER_REC ((x, tyx), m, ((y, cod), (u, w, dom)))
-       | CAP (dir, tubes, HOLE) => Syn.into @@ Syn.CAP {dir = dir, tubes = tubes, coercee = m}
-       | VPROJ (x, HOLE, f) => Syn.into @@ Syn.VPROJ (VarKit.toDim x, m, f)
-  in
-    fun unload (m || (syms, stk)) = 
-      case stk of
-         [] => m
-       | k :: stk => unload @@ plug m k || (syms, stk)
-  end
+  fun plug m =
```

Comment:
Any reason not to expose `plug`?

Reply:
feel free!

Reply:
Yay! The new order requires you to formally approve the PR, though.